doc/api/stream.md

```diff
@@ -2502,6 +2502,9 @@ and async iterators are provided below.
 })();
 ```
 
+Async iterators register a permanent error handler on the stream to prevent any
+unhandled post-destroy errors.
```

Comment:
Not sure that what I"m about to suggest is a good idea, so I'll defer to everyone else's judgment:

Is it worth re-wording to make it clear that this means the  `destroy()` method?

```suggestion
unhandled errors after `destroy()` executes.
```

I'm not sure that's an improvement to be honest, but maybe?